src/creditor/creditor.service.ts

```diff
@@ -32,6 +32,39 @@ export class CreditorService {
     return docRef;
   }
 
+  async checkId(
+    id: string,
+  ) {
+    const docRef = this.firebaseRepository.db
+      .collection(creditorCollection)
+      .doc(id);
+    const doc = await docRef.get();
+    if (!doc.exists) {
+      return null;
+    }
+    return docRef;
+  }
+
+  async createWithId(createCreditorDto: CreateCreditorDto, id: string): Promise<Creditor> {
+    // TODO: handle email or something already exists?
+    try {
+      const docRef = await this.firebaseRepository.db
+        .collection(creditorCollection)
+        .doc(id);
+      await docRef.set({
+          ...createCreditorDto,
+          createdAt: Date.now(),
+          updatedAt: Date.now(),
+        });
+      const data = (await docRef.get()).data();
+      return { id: id, ...data } as Creditor;
```

Comment:
Does Google ID really work? You also have users registered with Line, Facebook, etc. So, there are many kinds of IDs, which would cause complexity.